weedcoco/utils.py

```diff
@@ -55,4 +55,4 @@ def add_collection_from_file(coco, collection_path):
 
 def get_image_average_hash(path):
     """Return an average hash of an image"""
-    return str(imagehash.average_hash(PIL.Image.open(path)))
+    return str(imagehash.average_hash(PIL.Image.open(path), hash_size=16))
```

Comment:
This will certainly reduce the rate of false positives, but it would be good to substantiate this by doing an analysis of the similarity of existing images. That involves basically doing:

```
X = np.asarray([imagehash.average_hash(PIL.Image.open(path)).hash.flatten()
                for path in all_image_paths_on_RDS)
distances = scipy.spatial.pdist(X, metric='hamming')
```
and then looking at the histogram/distribution of `distances`. Alternatively, plotting the seaborn.clustermap of X with hamming distance might be demonstrative.

Reply:
The distributions are as below:
Hash size = 8
![RDS_distribution_hashsize_8](https://user-images.githubusercontent.com/51065213/97519858-16c74100-19ee-11eb-9113-2c0e558924e1.png)
Hash size = 16
![RDS_distribution_hashsize_16](https://user-images.githubusercontent.com/51065213/97519863-18910480-19ee-11eb-9396-0394270b8ec0.png)
There are some directories where images are mainly black, which contribute the high similarity, but still there no identical images with both sizes of hashing.

Reply:
Excellent. Yes, I thought the segmentation masks would be the source of clashes. Happy to adopt a moderate hash, e.g. hash_size=10 (in practice that's 50% more specific a hash).

Reply:
Thanks for that, by the way! Makes me a bit more comfortable.